timefold-solver/src/test/java/com/baeldung/timefoldsolver/ShiftScheduleSolverUnitTest.java

```diff
@@ -0,0 +1,63 @@
+package com.baeldung.timefoldsolver;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import java.time.Duration;
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ai.timefold.solver.core.api.solver.Solver;
+import ai.timefold.solver.core.api.solver.SolverFactory;
+import ai.timefold.solver.core.config.solver.SolverConfig;
+import ai.timefold.solver.core.config.solver.termination.TerminationConfig;
+
+public class ShiftScheduleSolverUnitTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(ShiftScheduleSolverUnitTest.class);
+
+    @Test
+    public void solve() {
+        SolverFactory<ShiftSchedule> solverFactory = SolverFactory.create(new SolverConfig().withSolutionClass(ShiftSchedule.class)
+            .withEntityClasses(Shift.class)
+            .withConstraintProviderClass(ShiftScheduleConstraintProvider.class)
+            // For this dataset, we know the optimal score in advance (which is normally not the case).
+            // So we can use .withBestScoreLimit() instead of .withTerminationSpentLimit().
+            .withTerminationConfig(new TerminationConfig().withBestScoreLimit("0hard/0soft")));
+        Solver<ShiftSchedule> solver = solverFactory.buildSolver();
+
+        ShiftSchedule problem = loadProblem();
+        ShiftSchedule solution = solver.solve(problem);
+        assertThat(solution.getShifts().size()).isNotZero();
+        for (Shift shift : solution.getShifts()) {
+            assertThat(shift.getEmployee()).isNotNull();
+        }
```

Comment:
I was thinking more about asserting on the specific hard coded optimal result data that is expected, rather than just a structural assertion

Reply:
There are multiple optimal solutions, so that would be too strict.
I'll assert the optimal score to unsure it's an optimal result.